google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/debugger/ui/ProjectDebuggeeBinding.java

```diff
@@ -128,26 +126,44 @@ public CloudDebugProcessState getInputState() {
 
   public void setInputState(@Nullable CloudDebugProcessState inputState) {
     this.inputState = inputState;
-    if (this.inputState != null) {
-      projectSelector.setText(this.inputState.getProjectName());
+    if (this.inputState != null
+        && !Strings.isNullOrEmpty(this.inputState.getProjectName())
+        && !Strings.isNullOrEmpty(this.inputState.getUserEmail())) {
+      Long projectNumber = null;
+      if (!Strings.isNullOrEmpty(this.inputState.getProjectNumber())) {
+        projectNumber = Long.parseLong(this.inputState.getProjectNumber());
+      }
+      projectSelector.setSelectedProject(
+          CloudProject.create(
+              this.inputState.getProjectName(), // TODO(ivanporty) add separate project name/ID
+              this.inputState.getProjectName(),
+              projectNumber,
+              this.inputState.getUserEmail()));
+
+      // update the state here as well
+      refreshDebugTargetList(projectSelector.getSelectedProject());
```

Comment:
just wondering why this needs to happen now, but it wasn't needed previously?

Reply:
That's again that tricky `ProjectSelector` thing - old `ProjectSelector` had a text field which fired text changed events all the time - so when this code called `setText()`, `ProjectSelector` would immediately fire text changed event, and this class would receive event and update its state as if new project has been set. We don't generate events when `ProjectSelector` is updated with project explicitly so we update the panel explicitly which makes it easier to read.

Reply:
I see. Do you think it would make sense to update the project selector so that it fired an event when the project selector is updated programmatically via setSelectedProject?

Reply:
If you do think it makes sense, I think it would also make certain unit tests easier (I recall running into this previously).

Reply:
Currently the use case is (1) we set the project so project selector updates UI, and we already know what this project is. (2) we listen to project selection change made by user. 
If we fire event when we call `setSelectedProject` it means we would have to disconnect listeners in some cases - otherwise it could be a loop of 'project changed' -> 'update' -> 'change project' -> listener again etc.

Reply:
ok sounds good. nevermind then.